### PR TITLE
Backend insert data  to cart table and search cart by PK ID

### LINKED ISSUE #18: (backend)用戶可以將商品加入購物車

### PR DESCRIPTION
**Pull request by wj461**
Just search and insert so no any logic in cart_usecase now

### CODE REVIEW

File: db/create-customer.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE IF NOT EXISTS UserData(
```
Comment (wj461) [inline db/create-customer.sql]: When UserData table pr merge fix it.

File: db/insert-product.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+
+INSERT INTO product VALUES
+    ((default), (SELECT store_id from stores LIMIT 1), 'name', 'good food', 10, decode('DEADBEEF', 'hex'), 123,2);
```
Comment (nonorze-VOXX) [inline db/insert-product.sql]: 無三不成禮

File: db/insert-customer.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO UserData VALUES
+    (1, 'asd', 'pwd', 'dsa@gmail.com', '123456', '0912-1123', '2022-10-10 11:30:30', 111);
```
Comment (nonorze-VOXX) [inline db/insert-customer.sql]: 無三不成禮

File: backend/api/swagger.yaml [hunk 4]
```diff
@@ -41,6 +41,62 @@ paths:
...
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "500":
+          description: Unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
```
Comment (nonorze-VOXX) [inline backend/api/swagger.yaml]: request review @huhuhu0420

File: backend/cart/delivery/cart_handler.go [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+	c.JSON(200, &swagger.CartInfo{
+		CustomerID:      _cart.CustomerID,
+		ProductID:       _cart.ProductID,
+		StoreID:         _cart.StoreID,
+		ProductQuantity: _cart.ProductQuantity,
+	})
+}
+
+func sub(s string, n int) string {
```
Comment (nonorze-VOXX) [inline backend/cart/delivery/cart_handler.go]: request rename

File: db/insert-product.sql (hunk 2 above)
Comment (nonorze-VOXX) [inline db/insert-product.sql]: (SELECT store_id from stores LIMIT 1) will break something

File: backend/cart/repository/postgresql.go [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+		logrus.Error(err)
+		return nil, err
+	}
+	return cart, nil
+}
+
+func (p *postgresqlCartRepo) PostCart(ctx context.Context, cart *domain.Cart) (*domain.Cart, error) {
+	sqlStatement := `
+	INSERT INTO cart (product_quantity, customer_id, product_id, store_id) VALUES
+    ( $1, (SELECT id from UserData LIMIT 1 OFFSET $2), (SELECT product_id from product LIMIT 1 OFFSET $3), (SELECT store_id from stores LIMIT 1 OFFSET $4));
```
Comment (nonorze-VOXX) [inline backend/cart/repository/postgresql.go]: FIX : LIMIT 1 OFFSET $2

File: backend/cart/repository/postgresql.go [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+	`
+	_, err := p.db.Exec(sqlStatement, cart.ProductQuantity, cart.CustomerID, cart.ProductID, cart.StoreID)
+	if err != nil {
+		logrus.Error(err)
+		return nil, err
+	}
+
+	_cart := &domain.Cart{}
+	row := p.db.QueryRow("SELECT customer_id, product_id, store_id, product_quantity FROM cart WHERE customer_id = $1 AND product_id = $2 AND store_id = $3", add(cart.CustomerID, 1), add(cart.ProductID, 1), add(cart.StoreID, 1))
```
Comment (nonorze-VOXX) [inline backend/cart/repository/postgresql.go]: request sql string to a variable

**Review by nonorze-VOXX — CHANGES_REQUESTED** [review]

File: db/insert-userData.sql [hunk 8]
```diff
@@ -0,0 +1,8 @@
+INSERT INTO UserData VALUES
+    ((default), 'asd', 'pwd', 'dsa@gmail.com', '123456', '0912-1123', '2022-10-10 11:30:30', 111);
+
+INSERT INTO UserData VALUES
+    ((default), 'asd', 'pwd', 'asd@gmail.com', '123456', '0912-1123', '2022-10-10 11:30:30', 111);
+
+INSERT INTO UserData VALUES
+    ((default), 'asd', 'pwd', 'avg@gmail.com', '123456', '0912-1123', '2022-10-10 11:30:30', 111);
```
Comment (nonorze-VOXX) [inline db/insert-userData.sql]: INSERT INTO no need three times
can the data more diff ?
we will cant see the different of front of user name

File: db/insert-product.sql (hunk 2 above)
Comment (wj461) [reply db/insert-product.sql]: fix

**Comment by nonorze-VOXX** [comment]
docker-compose up no work

**Review by nonorze-VOXX — APPROVED** [review]
LGTM

**Pull request merged**